File: tervezes/uml_gen/uml_generator/EquipmentAxe.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace VeryGoodViroGame
{
    public class EquipmentAxe : Equipment
    {
        private bool sharp = true;
        public override void Use(Virologist target)
        {
            Logger.NewFunctionCall(this, "Use");
            if (sharp)
            {
                target.KillVirologist();
                sharp = false;
            }

            Logger.ReturnFunction();
        }
    }
}

File: VeryGoodViroGame/src/VeryGoodViroGame/EquipmentAxe.java
package VeryGoodViroGame;

public class EquipmentAxe extends Equipment
{
    private boolean sharp = true;
    
    public EquipmentAxe()
    {
        name = "Axe";
    }
    
    @Override
    public void Use(Virologist target)
    {
        Logger.NewFunctionCall(this, "Use");
        if(sharp)
        {
            target.KillVirologist();
            sharp = false;
        }
        Logger.ReturnFunction();
    }
}
